src/rhsm/config.py

```diff
@@ -143,32 +144,35 @@ def read(self, file_names: Optional[List[str]] = None) -> List[str]:
         :param file_names: list of configuration files
         :return: number of configuration files read
         """
-        # FIXME Instead of optional list, we can set default to [] instead
         if file_names is None:
             return super(RhsmConfigParser, self).read(self.config_file)
         else:
             return super(RhsmConfigParser, self).read(file_names)
 
     def save(self, config_file: Optional[str] = None) -> None:
         """Writes config file to storage."""
-        # FIXME config_file is not used
-        rhsm_conf_dir: str = os.path.dirname(self.config_file)
+        # Use self.config_file if config_file is not provided in method arguments.
+        config_file: str = config_file or self.config_file
+
+        rhsm_conf_dir: str = os.path.dirname(config_file)
 
         # When /etc/rhsm does not exist, then try to create it
         if os.path.isdir(rhsm_conf_dir) is False:
             os.makedirs(rhsm_conf_dir)
 
-        # FIXME Stop using tempfile like that
+        # Create a temporary file to write config data to it and
+        # rename the file to the expected config file name after successfully
+        # writing all config data.
```

Comment:
So far, so good. Next time it is ok to add link of the BZ bug report to the comment of code.